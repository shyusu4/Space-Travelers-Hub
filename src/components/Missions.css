.missions-table {
  border-collapse: collapse;
  margin: 2rem;
}

.mission-name {
  font-weight: bold;
  text-align: left;
  padding-left: 5px;
  font-size: 20px;
}

.mission-item:nth-child(odd) {
  background-color: #ececec;
}

th,
td {
  border: 1px solid gray;
}

.mission-btn {
  padding: 5px;
}

.add-mission,
.leave-mission {
  width: 120px;
  font-size: 15px;
  margin-top: 20px;
  margin-bottom: 20px;
  height: 40px;
  border-radius: 6px;
  cursor: pointer;
}

.add-mission {
  background-color: white;
  border-color: gray;
}

.leave-mission {
  color: white;
  border-color: red;
  background-color: red;
}

.mission-badge {
  font-weight: bold;
  text-transform: uppercase;
  font-size: 15px;
  width: 150px;
  text-align: center;
}

.active-member {
  padding: 5px 7px;
  background-color: #1fa5e6;
  border-radius: 5px;
  color: #fff;
}

.non-member {
  padding: 5px 7px;
  background-color: grey;
  border-radius: 5px;
  color: #fff;
}

.mission-header > th {
  font-size: 28px;
  padding: 10px;
}

.mission-description {
  padding: 10px;
  line-height: 22px;
}
