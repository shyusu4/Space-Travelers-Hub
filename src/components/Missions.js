import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMissions } from '../redux/missions/missions';
import MissionItem from './MissionItem';
import './Missions.css';

function Missions() {
  const missions = useSelector((state) => state.missionsReducer.missions);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getMissions());
  }, [dispatch]);

  return (
      <table className="missions-table">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th></th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <MissionItem
              id={mission.id}
              key={mission.id}
              name={mission.name}
              description={mission.description}
            />
          ))}
        </tbody>
      </table>
  );
};

export default Missions;
