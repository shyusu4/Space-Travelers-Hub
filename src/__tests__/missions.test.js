import React from 'react';
import TestRender from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../components/Missions';
import store from '../redux/configureStore';

describe('Test Missions ', () => {
  it('Test Missions component ', () => {
    const component = TestRender.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
