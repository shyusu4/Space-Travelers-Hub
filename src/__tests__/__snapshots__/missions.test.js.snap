// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions  Test Missions component  1`] = `
<table
  className="missions-table"
>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th />
    </tr>
  </thead>
  <tbody />
</table>
`;
